pkg/driverbuilder/builder/vanilla.go

```diff
@@ -91,10 +91,15 @@ func (v vanilla) Script(c Config) (string, error) {
 		return "", err
 	}
 
-	kv := kernelrelease.FromString(c.Build.KernelRelease)
-
-	// Check (and filter) existing kernels before continuing
-	urls, err := getResolvingURLs([]string{fetchVanillaKernelURLFromKernelVersion(kv)})
+	kv := kernelReleaseFromBuildConfig(c.Build)
+
+	var urls []string
+	if c.KernelUrls == nil {
```

Comment:
For me, `kernelurls` is to pass the exact url for downloading a kernel-devel package (in case of Ubuntu, we need 2 urls, the arch one and the `_all` one).

Therefore there is no need to append it.  

Idea is that it is much simpler for an user to search for the exact url for its kernel devel, than to possibly adjust kernels scraping code inside driverkit, thus even if driverkit does not support discovering your url, you can still build your driver.
Moreover, this way we are able to support:
* various ubuntu flavors, like ubuntu-kvm, ubuntu-gcp and so on
* ubuntu/debian kernelversions like `32~16.04`

Note that this is part of a broader scope together with https://github.com/falcosecurity/kernel-crawler, that outputs a driverkit-like json for any found kernel with the urls to download its devel packages; see: https://github.com/falcosecurity/kernel-crawler/blob/main/kernels/x86_64/list.json

Reply:
What i am trying to say is that in the future users can try their luck by providing exact kernel url parameter whenever a "kernel header not found" error happens; and in case, fixing their configs to provide the `kernelurls`.

In the end, the current kernel url auto-formatting method will remain as a fallback for when no kernelurls param is passed.

That's of course IMHO :)

Reply:
> Idea is that it is much simpler for an user to search for the exact url for its kernel devel, than to possibly adjust kernels scraping code inside driverkit, thus even if driverkit does not support discovering your url, you can still build your driver.

👆 Definitely. This is a powerful addition!

The case I'm thinking of is, deploying Falco to a fleet of machines that has many instances of "kernel headers not found" because they have many e.g. ubuntu-gcp machines, but with different kernel versions. So, we know the kernel headers for ubuntu-gcp can all be found at e.g. `pub.ubuntu.com/blah/gcp/packages/kernel-devel-1.2.3-4.deb`. Could we have `kernelurls` accept the directory to `pub.ubuntu.com/blah/gcp/packages` and have it crawl that directory via the usual means, auto-extracting the URLs to kernel-devel packages it finds there?

Reply:
I see; perhaps that would be useful to also use a privare deb repository (for example).

I think it is a great idea; we might think of something like:
* if `kernelurls` contains some formatting options, like `%M.%m.%p` -> we format it with proper variables and use it 
* if `kernelurls` does not contain any formatting option, it means that it is a fullpath and we can use it without further effort

This is a nice solution, even if i'll have to write some formatting specs (like, in the above example: "%M" -> major, "%m" -> minor and "%p" -> patch).

What do you think?
Btw i can also try to split this feature from this PR if you wish; it helped me test various arm64 builds but it is not strictly related to it.

Reply:
> if `kernelurls` contains some formatting options, like `%M.%m.%p` -> we format it with proper variables and use it

Ahh, I think I see what you mean. So, for e.g. in the previous case mentioned with a fleet of `ubuntu-gcp` machines, one would do something like:

`driverkit --kernelurls pub.ubuntu.com/blah/ubuntu-gcp/packages/kernel-devel-%M.%m-%p.deb`

If so, this is a great idea! Allowing the user to specify the package URL format at runtime protects us against having to constantly update the builders if/when a vendor changes the formatting of their packages.

Doesn't necessarily need to be separate from this PR - I think it makes sense to include it here. We'll get this merged and then cut a new release. 🚀